src/app/(tabs)/search/index.tsx

```diff
@@ -105,6 +130,27 @@ function SearchScreen() {
     }
   };
 
+  const recentlyViewedStacking = (story: StoryPreview) => {
+    const maxArrayLength = 5;
+    const newRecentlyViewed = recentlyViewed;
+
+    for (let i = 0; i < recentlyViewed.length; i++) {
+      if (story.id === recentlyViewed[i].id) {
+        newRecentlyViewed.splice(i, 1);
+        break;
+      }
+    }
+
+    if (newRecentlyViewed.length >= maxArrayLength) {
+      newRecentlyViewed.splice(-1, 1);
+    }
+
+    newRecentlyViewed.splice(0, 0, story);
+
+    setRecentStory(newRecentlyViewed);
+    setRecentlyViewed(newRecentlyViewed);
```

Comment:
This [StackOverflow post](https://stackoverflow.com/questions/56266575/why-is-usestate-not-triggering-re-render) explains why the order isn't updating when you click on a recently viewed story. Basically, react doesn't see the list as having changed because the elements within it are still the same. By spreading it, we essentially tell it that it has changed and the useState causes a rerender as we want it to.

So, change this line to `setRecentlyViewed([...newRecentlyViewed])` and we should be good.